packages/devtools_app/lib/src/screens/memory/framework/memory_controller.dart

```diff
@@ -65,19 +68,46 @@ class MemoryFeatureControllers {
 ///
 /// The controller should be recreated for every new connection.
 class MemoryController extends DisposableController
-    with AutoDisposeControllerMixin {
+    with
+        AutoDisposeControllerMixin,
+        OfflineScreenControllerMixin<MemoryScreenOfflineData> {
   MemoryController({
     DiffPaneController? diffPaneController,
     ProfilePaneController? profilePaneController,
   }) {
-    controllers = MemoryFeatureControllers(
-      diffPaneController,
-      profilePaneController,
+    isOffline =
+        offlineController.shouldLoadOfflineData(ScreenMetaData.memory.id);
+    if (!isOffline) {
+      _controllers = MemoryFeatureControllers(
+        diffPaneController,
+        profilePaneController,
+      );
+    }
+  }
+
+  late final bool isOffline;
```

Comment:
we should not store this value on the memory controller. it may get out of sync. Use the global offlineController as the source of truth

Reply:
The only way to guarantee consistency is by using the `offlineController.offlineMode` value notifier. This is the source of truth for offline mode. Any piece of code that needs to check offline mode should check `offlineController.offlineMode.value`. If you search the codebase for "offlineController.offlineMode.value", you will see a lot of uses of this you can use for examples.

Reply:
Made memory controller listening for offlineController.offlineMode, and widget listening for initialization status of the controller. 
How does it look?